[Fed-Aero]-[Docs] Update the remaining README files (#3043)
open-edge-platform/edge-ai-suites: federal-aerospace/apps/handheld-multi-modal/README.md

SPDX-FileCopyrightText: (C) 2026 Intel Corporation
SPDX-License-Identifier: Apache-2.0
-->
# Handheld Multi-Modal Application

The Handheld Multi-Modal application is a full-stack AI inference and observability platform for handheld scenarios, optimized for Intel® edge hardware.

> The application combines LLM inference capability served through the OpenVINO™ Model Server platform, speech-to-text transcription through the Whisper service, a chat UI through the Open WebUI software, and metrics information through the Grafana dashboard; and runs with the [Visual Pipeline and Platform Evaluation Tool](https://github.com/open-edge-platform/edge-ai-libraries/tree/main/tools/visual-pipeline-and-platform-evaluation-tool) for pipeline visualization, sharing its Docker network.

Reviewer: ```suggestion
The application combines LLM inference capability served through the OpenVINO™ Model Server platform, speech-to-text transcription through the Whisper service, a chat UI through the Open WebUI software, and metrics information through the Grafana dashboard; it runs with the [Visual Pipeline and Platform Evaluation Tool](https://github.com/open-edge-platform/edge-ai-libraries/tree/main/tools/visual-pipeline-and-platform-evaluation-tool) for pipeline visualization, sharing its Docker network.
```
Maybe like this?

## Project Structure

```

> docker-compose.yml            Defines and runs all services in the application using the standard GPU configuration.

Author: @pperycz - Are the definitions for docker-compose.yml and docker-compose-cdi.yml correct?

|---------|-------|------|
| `grafana` | `grafana/grafana:latest` | Dashboards — consumes metrics via Grafana Live |
| `ovms` | `openvino/model_server:latest-gpu` | LLM inference via OpenAI-compatible REST API |
| `open-webui` | `ghcr.io/open-webui/open-webui:v0.9.6-slim` | Chat UI connected to OpenVINO model server |
| `whisper-stt` | `whisper-stt:latest` (local build) | Speech-to-text with Prometheus metrics |
| `nginx-https` | `nginx:alpine` | HTTPS reverse proxy (self-signed cert, enables browser microphone) |

All services share the `fedaero` Docker network and are defined in [`docker-compose.yml`](docker-compose.yml).

## Prerequisites

| Requirement | Notes |
|-------------|-------|
| Docker Engine release 24 or later| [Install guide](https://docs.docker.com/engine/install/) |
| Docker Compose v2 | Included with Docker Desktop; on Linux OS, install the `docker-compose-plugin` package. Use `docker compose` (space), not `docker-compose` (hyphen). |
| `render` group | Required for GPU access by OpenVINO model server. Verify with `getent group render`. |
| Git | Required for `make vippet-get` to do sparse-checkout of Visual Pipeline and Platform Evaluation Tool. |

> | Intel® GPU with OpenVINO driver | iGPU or discrete Intel® Iris® Xe MAX GPU supported. |

Author: @pperycz - Is "Intel® Iris® Xe MAX GPU " correct in this context?

Reviewer 2: It should be broader - this prerequisite section should cover all discrete GPUs supported by the driver, it's not limited to just Intel® Iris® Xe MAX GPU

Author: Thanks, @pperycz. I have rewritten the sentence to "iGPU or discrete GPU based on the Xe architecture.".

| **CDI only:** CDI drivers configured | Set `CDI_XE_DEVICE_0` in `.env` (see `AGENTS.md` for all env vars). |

> Run `make setup` after cloning to auto-detect the `render` the group GID and write the `.env`.

Reviewer 3: ```suggestion
Run `make setup` after cloning to auto-detect the `render` group GID and write the `.env`.
```

```bash
cd apps/handheld-multi-modal
make deploy        # standard GPU
make deploy-cdi    # CDI / SR-IOV environments
```

**Development or testing without Visual Pipeline and Platform Evaluation Tool** — a standalone variant creates its own isolated `fedaero` network. Metrics from Visual Pipeline and Platform Evaluation Tool pipelines will not be available.

```bash
make up-standalone
```

## Endpoints

Open WebUI, Grafana dashboard, and Whisper speech-to-text service are only accessible via the NGINX Transport Layer Security (TLS) reverse proxy — their container ports are not exposed directly to the host.

| Service | URL | Notes |
|---------|-----|-------|
| Open WebUI | https://localhost:8443 | LLM chat UI — browser microphone enabled (via NGINX reverse proxy) |
| Grafana dashboard | https://localhost:7443 | Pre-provisioned dashboards (via NGINX reverse proxy) |
| Whisper speech-to-text service | https://localhost:5443 | Speech-to-text — browser microphone enabled (via NGINX reverse proxy) |
| Visual Pipeline and Platform Evaluation Tool UI | https://localhost:443 | via NGINX reverse proxy |

## Make Targets

```

> make deploy            Full one-shot deploy (setup, Visual Pipeline and Platform Evaluation Tool, and this stack)

Reviewer 3: ```suggestion
make deploy            Full one-shot deploy (setup Visual Pipeline and Platform Evaluation Tool, and this stack)
```

Author: @jakubsikorski - Thanks for the suggestion. Based on the original sentence, I believe we are talking about three things here: the setup, the Visual Pipeline and Platform Evaluation Tool, and the stack. Your suggestion made me realize that we should make things clearer - I'll check with the subject matter expert.

Author: @pperycz - would the following definition be clearer for "make deploy"?

full one-shot deployment: detects the render group GID and writes it into the .env file (idempotent), sets up Visual Pipeline and Platform Evaluation Tool, and sets up this stack.

Reviewer 2: ```suggestion
make deploy           # full one-shot deployment: sets runtime environment variables for the stack, tailors Visual Pipeline and Platform Evaluation Tool installation (metrics-manager and supported models) and brings up the stack components in the correct order
```
